mqtt/mqtt-broker/src/broker.rs

```diff
@@ -100,6 +104,44 @@ where
         Ok(self.snapshot())
     }
 
+    fn reevaluate_subscriptions(&mut self) -> Result<(), Error> {
+        let mut client_ids_to_remove: Vec<ClientId> = Vec::new();
+        for session in &self.sessions {
```

Comment:
you can deconstruct the item right away to avoid things like this `session.1...`
```suggestion
       for (client_id, session) in &self.sessions {
```